frontend/src/metabase/components/InputBlurChange.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState, useEffect } from "react";
+import _ from "underscore";
+import Input, { InputProps } from "metabase/core/components/Input";
+// import { Input } from "./InputBlurChange.styled";
+
+/**
+ * A small wrapper around <input>, primarily should be used for the
+ * `onBlurChange` feature, otherwise you should use <input> directly
+ */
+
+interface InputBlurChangeProps extends InputProps {
+  onBlurChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+const InputBlurChange = ({
+  value,
+  onChange,
+  onBlurChange,
+  ...props
+}: InputBlurChangeProps) => {
+  const [internalValue, setInternalValue] = useState(value);
+
+  useEffect(() => {
+    setInternalValue(value);
+  }, [value]);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInternalValue(event.target.value);
+    if (onChange) {
+      onChange(event);
+    }
+  };
+
+  const handleBlur = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (onBlurChange && (value || "") !== event.target.value) {
+      onBlurChange(event);
+    }
+  };
+
+  const inputProps = _.omit(props, "onBlur");
```

Comment:
btw would it be better to call the original `onBlur` in `handleBlur` rather than omit it?